### ISSUE TITLE
IRS Planning - Save Plan as a Draft

### ISSUE THREAD

**Issue by cKellyDesign** [onaio/reveal-frontend#141]
While planning the user needs to be capable of Saving a Plan as a non-finalized draft.

#### Dependencies
- ~~Non-client Store for the Plan definition data - @moshthepitt @samwata @craigappl let's discuss this point asap~~

#### Requirements
- [x] On all Saves, render feedback indicating Save success/fail
- [ ] On first Save, the webapp waits for server confirmation before navigating to `draft`
- [x] On first Save, the new Plan will be rendered when navigating to `/irs`

#### Interactions
- [ ] Clicking the `Save as a draft` button:
    - [x] Plan configuration is POST/PUT to OpenSRP or a Store
    - [ ] If saving from `create`: navigate to `/irs/draft/<plan-id>`
    - [x] Else: don't navigate

**Comment by moshthepitt** [comment]
When creating plan goals/targets, the goal comparator should be limited to "=", ">=" or ">"

**Comment by moshthepitt** [comment]
Note that this is blocked by #156 because we do not have anywhere to store drafts

**Comment by cKellyDesign** [comment]
> Note that this is blocked...

@moshthepitt Agreed. We need a lightweight storage solution we can implement asap.

Alternatively, for the initial planning release (july 30th) we could skip the Draft step since it was not originally scoped and the planning flow can be achieved without it. Then when we have the storage / persisting credentials working we could iterate and add that step of the process.

cc @mberg @craigappl

**Comment by moshthepitt** [comment]
Yes we most probably do need to skip the draft for now

**Comment by cKellyDesign** [comment]
> Yes we most probably do need to skip the draft for now

@moshthepitt - @mberg mentioned that the client probably won't be too happy with this since the process of planning takes quite a bit of time. :-/

**Comment by moshthepitt** [comment]
@cKellyDesign @mberg  We can commit to doing it, but not sure we shall make it by July 31, but we can certainly try.

**Comment by moshthepitt** [comment]
@cKellyDesign so, after @mberg made this comment right here https://github.com/onaio/reveal-frontend/issues/188#issuecomment-511625319 it occurred to me that we can just simply save the draft plans by sending them to OpenSRP.  The caveat is, I guess, the plan will have to be complete "enough" for it to pass OpenSRP's validation(s).

If we choose to go this way, then we'll be unblocked.

What are your thoughts?


cc: @githengi

**Comment by mberg** [comment]
That’s the idea. And we will need to be able to oldster to go from draft to
active

**Comment by cKellyDesign** [comment]
@moshthepitt - I'm not really sure what that means in terms of OpenSRP (like what is stored and where). If we are able to PUSH/PUT/GET into/from OpenSRP endpoints then this sounds ideal.

@githengi - Can you comment on the feasibility of this approach? Aka - can we save a Plan entity as a draft without generating all the Tasks? And can we distinguish drafts from finalized Plans?

**Comment by githengi** [comment]
Yes @cKellyDesign @moshthepitt @mberg plans can be saved as draft on OpenSRP. These plans can be update progressively until they become activate and later tasks generated

 Plans don't have to have tasks that can be done later.

**Comment by cKellyDesign** [comment]
@githengi Can Plans also be deleted via OpenSRP endpoints?

**Comment by cKellyDesign** [comment]
In order to connect directly to the OpenSRP endpoints we will need to add an OpenSRP service module and possibly a connector module similar to [superset service](https://github.com/onaio/reveal-frontend/blob/master/src/services/superset/index.ts) and the [SupersetConnector](https://github.com/onaio/js-tools/tree/master/packages/SupersetConnector).

cc - @moshthepitt

**Comment by cKellyDesign** [comment]
@mberg @craigappl what are the hardcoded IRS plan goals? Do we need actions?

**Comment by moshthepitt** [comment]
@cKellyDesign I'll await @craigappl to provide more light on this but generally we want to always have both goals and actions for any activity.

The [web form spec](https://docs.google.com/spreadsheets/d/1kGPgyF0eJdag19FFL1Oog2gdphdPtKobY4EQ0gMgtgw/edit#gid=1805975071) seems to indicate that IRS plans have only one goal-action pair though

**Comment by cKellyDesign** [comment]
We need to add: 
* visual confirmation of success (popup modal? alert banner?)
* visual warning of failure (popup modal? alert banner?)
* re-direct on confirmation of successful save from OpenSRP

**Comment by cKellyDesign** [comment]
Now tracking at https://github.com/onaio/reveal-frontend/issues/308
